src/Windows/Avalonia.Direct2D1/Avalonia.Direct2D1.csproj

```diff
@@ -25,4 +26,5 @@
   <Import Project="..\..\..\build\SharpDX.props" />
   <Import Project="..\..\Shared\RenderHelpers\RenderHelpers.projitems" Label="Shared" />
   <Import Project="..\..\..\build\JetBrains.Annotations.props" />
+  <Import Project="..\..\..\build\Unsafe.props" />
```

Comment:
Are we adding a nuget package dependency just to replace kernel32!CopyMemory?

Reply:
kernel32!CopyMemory doesn't exist on x64. And there are other places we can use System.Runtime.CompilerServices.Unsafe. Also this library is extremely small (one class with each method written in IL with like 5 instructions max).

Reply:
I've just checked and it *does* exist on x64. We need to change the signature to use `UIntPtr` instead of `uint` though.

Reply:
When I tried I got a DllNotFoundException. That's why I made this change in the first place.

Reply:
![default](https://user-images.githubusercontent.com/1067584/34268709-cb1abcfa-e692-11e7-8aae-a6205861bcaf.png)

Reply:
Weird. I'll try again on my machine, but without a change it crashed in our build script the last time I tried.

Reply:
The reason I don't want to have dependencies on BCL-related stuff is NuGet's idiotic nearest-wins policy which might ruin someone's day because of that.

Reply:
![2017-12-21](https://user-images.githubusercontent.com/1571408/34279144-b8c910f6-e674-11e7-996a-5d9b2e5f71bf.png)
It's failing on the build script here. Take a look:

Reply:
The API of `System.Runtime.CompilerServices.Unsafe` is pretty stable and unlikely to change/add methods. Of all the BCL-related packages, this one is the safest to reference.

Reply:
Please, try to use 
```
 [DllImport("msvcrt.dll", SetLastError = false)]
public static extern IntPtr memcpy(IntPtr dest, IntPtr src, IntPtr count); 
``` 
instead.

I'm not concerned about API stability, I'm concerned about "detected package downgrade" nonsense.

Reply:
I'll try it, but I think using `System.Runtime.CompilerServices.Unsafe` is worth the (two second) developer annoyance of having to add a direct (instead of multiple transitive) package dependency.

It's more performant to not have to invoke all of the PInvoke machinery for something we can keep in the CLR.

Reply:
The "detected package downgrade" situation isn't even a complete build break without a workaround. It's really easy to work around the situation.

Reply:
The `memcpy` solution works. I can put that in if you'd still prefer to use that over staying within the CLR.

Reply:
> isn't even a complete build break without a workaround

You see, I kinda want people to have Avalonia working out of the box. Without having to figure out exact package versions that they need to manually install.

So, for now, please, use `memcpy`.

Reply:
Ok :+1:

Reply:
Fixed! Anything else you'd like me to change?